### PR TITLE
Allow the main process to run Chapel task bodies.

### PR DESCRIPTION
**Pull request by gbtitus**
We can't run a Chapel task body unless we have Qthreads thread (task)
local storage, because that's where the serial state and sublocale info
are stored.  But only actual qthreads have Qthreads TLS.  The comm layer
polling thread doesn't, for example.  However, we provide fake TLS for
the comm thread and return that in chpl_qthread_get_tasklocal(), so that
the comm thread AM handler can execute "fast" task bodies directly.

Here we add the same support for the main process on each locale, since
at least for non-0 locales that process can end up doing AM handling in
certain comm layers (notably GASNet) as a result of calling into
chpl_comm_barrier() to handle exit.

This fixes 3 intermittent SIGABRT failures in
  optimizations/sungeun/optimized-on/test_OptimizedOn_Block1
resulting from commit 870da1a80cde31c30d3dc49a83313e1845b6ae3f.

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho, can you look at this one?

File: third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.h [hunk 1]
```diff
@@ -307,14 +309,17 @@ void chpl_task_stdModulesInitialized(void);
...
-    chpl_qthread_tls_t * tls;
+    chpl_qthread_tls_t* tls;
 
     if (chpl_qthread_done_initializing) {
         tls = (chpl_qthread_tls_t *)
               qthread_get_tasklocal(sizeof(chpl_qthread_tls_t));
         if (tls == NULL &&
             pthread_equal(pthread_self(), chpl_qthread_comm_pthread))
-          tls = &chpl_qthread_comm_task_tls;
+            tls = &chpl_qthread_comm_task_tls;
```
Comment (ronawho) [inline third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.h]: Is it worth changing this to:

 if (tls == NULL) {
  if (pthread_equal(pthread_self(), chpl_qthread_comm_pthread))
    tls = &chpl_qthread_comm_task_tls;
  else if (pthread_equal(pthread_self(), chpl_qthread_process_pthread))
    tls = &chpl_qthread_process_tls;
}

to eliminate the second NULL comparison?

**Comment by ronawho** [comment]
Looks good to me, up to you if you want to change the code I commented on.

File: third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.h (hunk 1 above)
Comment (gbtitus) [inline third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.h]: Yep, that's a good idea and I've done it and re-tested.

**Pull request merged**